deep4cast/forecasters.py

```diff
@@ -1,28 +1,24 @@
-# -*- coding: utf-8 -*-
+from inspect import getfullargspec
 import time
-import torch
+from typing import Union
 
 import numpy as np
-
-from typing import Union
-from inspect import getfullargspec
+import torch
 
 
 class Forecaster():

```

Comment:
What is a `DataLoader`? It sounds too specific to `PyTorch`. Users should not need to learn this new concept in order to use D4C. 

Suggest changing `DataLoader` to `DataFrame` if possible. All valid data scientists know `DataFrame`. Very few (including me) know `DataLoader`.

Reply:
Why do we have both `fit` and `train` in `Forecaster`?

Reply:
1. `Dataloader`: `Datasets` and `Dataloader` go hand-in-hand, see [here](https://pytorch.org/docs/stable/torchvision/datasets.html). It is indeed specific to `pytorch`, but so is all the code in this branch. A demo of how to use them is in the example notebook.

2. `fit` and `train`: We use `fit` to bring the API closer to `sklearn`. `train` is only meant to be called by `fit`, I suggest we change it to a protected function.